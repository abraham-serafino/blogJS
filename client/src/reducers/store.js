// @flow
import { createStore, combineReducers } from 'redux';
import AppReducer from './AppReducer.js';

export default createStore(combineReducers({
  appState: AppReducer
}));
