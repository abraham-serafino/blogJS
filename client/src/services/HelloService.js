// @flow
const HelloService = {
  getHello(): Promise<*> {
    return fetch('http://localhost:9090/api/v1/hello')
        .then((response: Response) => response.json());
  }
};

export default HelloService;
