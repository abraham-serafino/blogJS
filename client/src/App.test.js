// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import store from './reducers/store.js';
import HelloService from './services/HelloService.js';
import AppComponent from './App.js';

describe('App', () => {
  it('should render without crashing', () => {
    const service = HelloService;
    const App = AppComponent({ service });

    const component = renderer.create(
      <Provider store={store}><App/></Provider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
