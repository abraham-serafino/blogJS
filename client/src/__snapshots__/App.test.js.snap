// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<div>
  <h2>
    Welcome to My Site
  </h2>
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Change title
  </button>
</div>
`;
